realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -215,10 +215,13 @@ JNIEXPORT jboolean JNICALL Java_io_realm_internal_Table_nativeIsColumnNullable(J
 // 4d. the column to be converted will index shifted one place to column_index + 1
 // 5. search indexing must be preserved
 // 6. removing the original column and renaming the temporary column will make it look like original is being modified
+//
+// WARNING: These methods do NOT work on primary key columns if the Realm is synchronized.
```

Comment:
maybe add an assertion that throws if called from sync
```C++
#if REALM_ENABLE_SYNC
REALM_ASSERT(false);
#endif
```

Reply:
Good idea 👍